File: Back/.Net/BetFriends.Api/Features/Bets/RetrieveBetsController.cs
using BetFriends.Bets.Application.Abstractions;
using BetFriends.Bets.Application.Features.RetrieveBets;
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

namespace BetFriends.Api.Features.Bets
{
    public class RetrieveBetsController(IBetModule betModule) : Controller
    {
        private readonly IBetModule betModule = betModule;

        [SwaggerOperation(Tags = new[] { "Bets" })]
        [HttpGet("bets")]
        public async Task<IActionResult> RetrieveBetsAsync()
        {
            var bets = await betModule.ExecuteAsync(new RetrieveBetsQuery());
            return Ok(bets);
        }
    }
}

File: Back/NodeJS/src/infrastructure/api/nest/src/bet/features/retrieve-bets/RetrieveBets.controller.ts
import { Controller, Get, HttpStatus, Inject, Res } from "@nestjs/common";
import { IBetModule } from "../../../../../../../application/Abstractions/IBetModule";
import { FastifyReply } from 'fastify';
import { RetrieveBetsQuery, RetrieveBetsResponse } from '../../../../../../../application/features/retrieve-bets/RetrieveBetsQueryHandler';

@Controller('bets')
export class RetrieveBetsController {
    constructor(@Inject('IBetModule') private betModule: IBetModule) {}

    @Get()
    async RetrieveBets(@Res() res: FastifyReply) {
        const bets = await this.betModule.ExecuteQuery<RetrieveBetsResponse[]>(new RetrieveBetsQuery());
        return res.code(HttpStatus.OK).send(bets);
    }
}